src/Catalog/Telemetry/TelemetryService.cs

```diff
@@ -91,5 +64,10 @@ public void TrackCatalogIndexWriteDuration(TimeSpan duration, Uri uri)
                     { Uri, uri.AbsoluteUri },
                 });
         }
+        
+        public virtual DurationMetric TrackDuration(string name, IDictionary<string, string> properties = null)
```

Comment:
DurationMetric is a class Loic added in NuGet.Services.Logging. Very useful. Given more time I would add the ability to add/set properties at runtime. Right now the list of properties is given in ctor, and in order to modify them u need to maintain a reference to the dictionary.